### ISSUE TITLE
Event handling issues on macOS

### ISSUE THREAD

**Issue by parasyte** [fasihrana/skryn#24]
I have a couple of issues on macOS:

* Events only trigger when the cursor is over the edge of the window (when the cursor shape turns into a drag icon). This is easy to see with the calculator example.
* Solid white window while resizing the window.

macOS does have some special consideration for rendering while resizing. In short, the rendering must be performed in the resize handler. See: https://github.com/tomaka/glutin/issues/445

If you were to split `Window.tick` into two methods, one that handles tick events and one that renders, then the render method could be called from the `WindowEvent::Resized` handler. It looks like you tried to do that with `Window.render_root`. But there is a lot of additional stuff needed like building display lists, and updating the GL state for glutin. That's about 1/3 of what `Window.tick` does.

**Comment by fasihrana** [comment]
Bummer about macOS resizing. tomaka/winit#39 seems to have been closed though. Will look into this in detail.

**Comment by parasyte** [comment]
Yes, the winit issue is resolved, but it's still left up to the app to do rendering in the resize callback.

**Comment by fasihrana** [comment]
On macOS, CursorEntered and CursorLeft are fired at the same time. Possibly related issue tomaka/winit#807
